cmd/gcp-controller-manager/node_annotator.go

```diff
@@ -286,6 +309,51 @@ func extractKubeLabels(instance *compute.Instance) (map[string]string, error) {
 	return parsedLabels, nil
 }
 
+func extractNodeTaints(instance *compute.Instance) ([]core.Taint, error) {
+	const kubeEnvKey = "kube-env"
+
+	if instance.Metadata == nil {
+		return nil, errNoMetadata
+	}
+
+	var kubeEnv *string
+	for _, item := range instance.Metadata.Items {
+		if item == nil || item.Key != kubeEnvKey {
+			continue
+		}
+		if item.Value == nil {
+			return nil, fmt.Errorf("instance %q had nil %q", instance.SelfLink, kubeEnvKey)
+		}
+		kubeEnv = item.Value
+	}
+	if kubeEnv == nil {
+		return nil, errNoMetadata
+	}
+	if len(*kubeEnv) == 0 {
+		return nil, nil
```

Comment:
is this a happy path? Would it confuse some future eng that the taints logic below isn't completing because (not obvious to them) the kubeEnv is empty? If so maybe make this an error. If not then its fine :)

Reply:
Yes, this is a happy path with respect to merging taints. But to make sure it is easier to understand what is going on, I have added a log line.